kevalpatel2106/pocket-ci
# Allow pinning of project

## Overview :notebook:




PR adds a new table _(and migrates DB to version 2)_ to store the persistent data related to the project. Currently, this data only includes information about whether the project is pinned or not.

#### Task left:
- [x] Change pin icon
- [x] Add instrumentation test for project local data dao


## Links to issues/other PRs :link:



- closes: #79 


## Screenshots, video :framed_picture:




https://user-images.githubusercontent.com/20060162/192146618-eafb5cdc-1cdb-498a-9512-a5a6d8ac62fd.mp4


## Testing :mag:



1. Load the project list
2. Pin any project
3. Verify the project list order changes to show pinned projects at the top.
4. Refresh the project list and verify the pinned project persist.


## Notes :spiral_notepad:



This PR migrates the DB to version 2. It includes migration and test for migration.

## Checklist :1234:



- [x] Feature smoke-tested
- [x] Included unit tests (if needed)
